src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -357,18 +348,229 @@ FESurfaceDistanceEvaluator::computeSignedDistance(int n_idx, int d_idx)
     TBOX_ASSERT(d_idx >= 0);
 #endif
 
-    if (d_use_vol_extracted_bdry_mesh)
+    // Loop over patches on finest level.
+    const int finest_ln = d_patch_hierarchy->getFinestLevelNumber();
+    Pointer<PatchLevel<NDIM> > level = d_patch_hierarchy->getPatchLevel(finest_ln);
+    for (PatchLevel<NDIM>::Iterator p(level); p; p++)
     {
-        computeSignedDistanceVolExtractedBdryMesh(n_idx, d_idx);
+        Pointer<Patch<NDIM> > patch = level->getPatch(p());
+        const Box<NDIM>& patch_box = patch->getBox();
+        const SAMRAI::hier::Index<NDIM>& patch_lower_index = patch_box.lower();
+        Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+        const double* patch_X_lower = patch_geom->getXLower();
+        const double* const patch_dx = patch_geom->getDx();
+        Pointer<CellData<NDIM, double> > n_data = patch->getPatchData(n_idx);
+        Pointer<CellData<NDIM, double> > d_data = patch->getPatchData(d_idx);
+
+        // Note that we only work with cells that satisfy the intersecting criteria.
+        for (Box<NDIM>::Iterator it(patch_box); it; it++)
+        {
+            CellIndex<NDIM> ci(it());
+            const bool contains_key = (d_cell_elem_neighbor_map.find(ci) != d_cell_elem_neighbor_map.end());
+            if (contains_key)
+            {
+                std::set<Elem*>& elem_set = d_cell_elem_neighbor_map[ci];
+                const int num_elements = static_cast<int>(elem_set.size());
+                (*n_data)(ci) = num_elements;
+
+                // Loop over the cutting elements and find the minimum distance.
+                IBTK::VectorNd P;
+                for (int d = 0; d < NDIM; ++d)
+                {
+                    P[d] = patch_X_lower[d] + patch_dx[d] * (static_cast<double>(ci(d) - patch_lower_index(d)) + 0.5);
+                }
+                double min_dist = std::numeric_limits<double>::max();
+
+                // Create a pair to take care of projection and pseudo-normal for cells
+                // equidistant to multiple elements.
+                std::vector<std::pair<IBTK::VectorNd, IBTK::VectorNd> > vec_equidistant_pair;
+
+                for (auto& elem : elem_set)
+                {
+                    IBTK::VectorNd v, w, proj;
+                    double dist = std::numeric_limits<double>::max();
+
+                    // Get the nodes.
+                    const libMesh::Point& n0 = elem->point(0);
+                    const libMesh::Point& n1 = elem->point(1);
+#if (NDIM == 3)
+                    const libMesh::Point& n2 = elem->point(2);
+#endif
```

Comment:
n0, n1, and n2 aren't used in 3D. Could you remove n2 and then move n1 and n2 inside the `#ifdef` block below?